nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -59,28 +65,51 @@ class ParametersForm(BootstrapMixin, forms.Form):
             required=True,
             help_text="The key of the HashiCorp Vault secret",
         )
+        hashicorp_vault = forms.ChoiceField(
+            required=False,  # This should be required, but would be a breaking change
+            choices=hashicorp_vault_choices,
+            help_text="HashiCorp Vault where secret is stored in.",
+        )
         mount_point = forms.CharField(
             required=False,
-            help_text=f"The path where the secret engine was mounted on (Default: <code>{DEFAULT_MOUNT_POINT}</code>)",
-            initial=DEFAULT_MOUNT_POINT,
+            help_text="Override Vault Setting: The path where the secret engine was mounted on (Default: <code>secret</code>)",
+            label="Mount Point (override)",
         )
         kv_version = forms.ChoiceField(
             required=False,
-            choices=HashicorpKVVersionChoices,
-            help_text=f"The version of the kv engine (either v1 or v2) (Default: <code>{DEFAULT_KV_VERSION}</code>)",
-            initial=DEFAULT_KV_VERSION,
+            choices=add_blank_choice(HashicorpKVVersionChoices),
+            help_text="Override Vault Setting: The version of the kv engine (either v1 or v2) (Default: <code>v2</code>)",
+            label="KV Version (override)",
         )
 
+    @staticmethod
+    def retrieve_vault_settings(name=None):
+        """Retrieve the configuration from settings that matches the provided vault name.
+
+        Args:
+            name (str, optional): Vault name to retrieve from settings. Defaults to None.
+
+        Returns:
+            vault_settings (dict): Hashicorp Vault Settings
+        """
+        vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"].get("hashicorp_vault", {})
+        if name and name != "default":
+            vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]["hashicorp_vault"]["vaults"].get(
+                name, {}
+            )
```

Comment:
What if I want to have a `vaults["default"]` entry? Maybe:

```suggestion
        vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"].get("hashicorp_vault", {})
        if name or "vaults" in vault_settings:
            return vault_settings["vaults"][name]
```

(I think it'd be better to go ahead and raise the KeyError if a nonexistent vault is selected, rather than quietly returning an empty dict)

Reply:
*Maybe* even treat a `vault_name` of `None` as equivalent to `"default"` to provide a migration path forwards?

Reply:
I still need to support legacy though... This is why there is `default`, I can rename it though.